CRM/Upgrade/Incremental/MessageTemplates.php

```diff
@@ -476,6 +476,15 @@ public function replaceTokenInGreetingOptions(string $old, string $new): void {
    */
   public function getUpgradeMessages() {
     $updates = $this->getTemplatesToUpdate();
+
+    // workflow_name is not available pre-5.26, so we won't check if templates are changed or not pre-5.26
+    try {
+      $uneditedTemplates = $this->getUneditedTemplates();
+    }
+    catch (Exception $e) {
+      $uneditedTemplates = [];
```

Comment:
Thanks for doing this.

I think the test problem is that it's enforcing what we're actually trying to fix. A better test might be to have it edit one of them, and then the result list should just be that one.

```
CRM_Upgrade_Incremental_BaseTest::testMessageTemplateGetUpgradeMessages
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    'Memberships - Receipt (on-line)' => 'Use email greeting at top whe...ilable'
-    'Contributions - Receipt (on-line)' => 'Use email greeting at top whe...ilable'
-    'Events - Registration Confirmation and Receipt (on-line)' => 'Use email greeting at top whe...ilable'
 )

/home/homer/buildkit/build/build-3/web/sites/all/modules/civicrm/tests/phpunit/CRM/Upgrade/Incremental/BaseTest.php:139
```

Reply:
Yes, have pushed a new version doing this.